pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/v2/BasePNConfiguration.kt

```diff
@@ -0,0 +1,528 @@
+package com.pubnub.api.v2
+
+import com.pubnub.api.UserId
+import com.pubnub.api.crypto.CryptoModule
+import com.pubnub.api.enums.PNHeartbeatNotificationOptions
+import com.pubnub.api.enums.PNLogVerbosity
+import com.pubnub.api.retry.RetryConfiguration
+import okhttp3.Authenticator
+import okhttp3.CertificatePinner
+import okhttp3.ConnectionSpec
+import okhttp3.logging.HttpLoggingInterceptor
+import java.net.Proxy
+import java.net.ProxySelector
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.SSLSocketFactory
+import javax.net.ssl.X509ExtendedTrustManager
+
+/**
+ * A storage for user-provided information which describe further PubNub client behaviour.
+ * Configuration instance contains additional set of properties which
+ * allow performing precise PubNub client configuration.
+ *
+ */
+interface BasePNConfiguration {
```

Comment:
What is  the value of having this interface and BasePNConfigurationImpl?

Reply:
just to follow the convention of having interfaces in core-api and implementations in core-impl really...
it also has some positive sides, for example lets us be less strict about visibility modifiers in core-impl classes, for example: if we need something visible for testing in `*Impl` we can just make it public because it won't be visible to SDK users anyway. Another side effect is that the build is faster when changes happen in *Impl classes only, especially when we further split java and kotlin into api/impl

Reply:
ok, thanks.